modules/shared/src/test/scala/org/tessellation/schema/BalanceSuite.scala

```diff
@@ -0,0 +1,43 @@
+package org.tessellation.schema
+
+import org.tessellation.schema.balance.{Amount, Balance}
+
+import eu.timepit.refined.types.numeric.NonNegLong
+import weaver.FunSuite
+import weaver.scalacheck.Checkers
+
+object BalanceSuite extends FunSuite with Checkers {
+  test("plus should add given amount to balance") {
+    val balance = Balance(NonNegLong(1L))
+    val amount = Amount(NonNegLong(2L))
+
+    val newBalance = balance.plus(amount)
+
+    val expected = Balance(NonNegLong(3L))
+
+    exists(newBalance)(expect.same(_, expected))
+  }
+
+  test("plus returns Left when sum overflows the Long range") {
+    val balance = Balance(NonNegLong(Long.MaxValue))
+    val amount = Amount(NonNegLong(2L))
+
+    not(exists(balance.plus(amount))(_ => success))
+  }
+
+  test("minus should subtract given amount from balance") {
+    val balance = Balance(NonNegLong(30L))
+    val amount = Balance(NonNegLong(1L))
+
+    val expected = Balance(NonNegLong(29L))
+
+    exists(balance.minus(amount))(expect.same(_, expected))
+  }
+
+  test("minus returns Left when amount lowers balance below zero") {
+    val balance = Balance(NonNegLong(1L))
+    val amount = Balance(NonNegLong(30L))
+
+    not(exists(balance.minus(amount))(_ => success))
```

Comment:
Looks a bit weird maybe you could just assert that a value is equal to `Left`?

Reply:
I can do that. anyway, `exists` work on `foldable` type (so on Either as well), and `not(exists` is imo self explanatory.